### PR TITLE
feat(protocol): add `AuthorizeTaikoForMultihop ` script

### PR DESCRIPTION
**Pull request by davidtaikocha**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new smart contract to enable multihop functionality for Taiko contracts, allowing for enhanced authorization management across multiple contracts.

### CODE REVIEW

File: packages/protocol/script/AuthorizeRelayer.s.sol [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+//   Blog: https://mirror.xyz/labs.taiko.eth
+//   Youtube: https://www.youtube.com/@taikoxyz
+
+pragma solidity 0.8.24;
+
+import "../test/DeployCapability.sol";
+import "../contracts/L1/gov/TaikoTimelockController.sol";
+import "../contracts/signal/SignalService.sol";
+
+contract AuthorizeRelayer is DeployCapability {
```
Comment (dantaik) [inline packages/protocol/script/AuthorizeRelayer.s.sol:21]: lets rename it to AuthorizeTaikoForMultihop

File: packages/protocol/script/AuthorizeRelayer.s.sol [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    uint256 public privateKey = vm.envUint("PRIVATE_KEY");
+    address public sharedSignalService = vm.envAddress("SHARED_SIGNAL_SERVICE");
+    address[] public relayers = vm.envAddress("RELAYERS", ",");
```
Comment (dantaik) [inline packages/protocol/script/AuthorizeRelayer.s.sol:24]: TAIKO_CONTRACTS

Comment (dantaik) [inline packages/protocol/script/AuthorizeRelayer.s.sol:24]:
```suggestion
    address[] public taikoContracts = vm.envAddress("TAIKO_CONTRACTS", ","); // TaikoL1 and TaikoL2 contracts
```

File: packages/protocol/script/AuthorizeRelayer.s.sol (hunk 1 above)
Comment (davidtaikocha) [reply packages/protocol/script/AuthorizeRelayer.s.sol]: Renamed

File: packages/protocol/script/AuthorizeRelayer.s.sol (hunk 2 above)
Comment (davidtaikocha) [reply packages/protocol/script/AuthorizeRelayer.s.sol]: Changed

Comment (davidtaikocha) [reply packages/protocol/script/AuthorizeRelayer.s.sol]: Changed

**Review by dantaik — APPROVED** [review]

**Review by dantaik — APPROVED** [review]

**Review by adaki2004 — APPROVED** [review]

**Comment by Uzzairedrix** [comment]
Great

**Pull request merged**